### DISCUSSION TITLE
How to convert between colour spaces?

### DISCUSSION THREAD

**Discussion by hminle** [colour-science/colour#735] [Q&A] [answered]
Hi Colour Team,

I want to convert from ProPhoto RGB to sRGB
I think of one way to do it: ProPhoto RGB --> XYZ --> sRGB
But the result seems to be wrong
![image](https://user-images.githubusercontent.com/16201681/73987166-7edc7100-490d-11ea-8eaf-b6c4df2e9811.png)

And another thing, I don't know how to choose ILLUMINANT values for the converting function.
Do you guys have any guide lines?

Best,
Hoang

**Comment by thomasmansencal** [comment, marked as answer]
The easiest way to do that is to use the `colour.RGB_to_RGB` definition, it will compute a transformation matrix with chromatic adaptation if needed (and required by the user).

Note that to get the ProPhoto RGB colourspace class instance, you don't have to do what you did with your first line of code. We do that in the `colour.plotting` sub-package because the filtering definitions accept regular expressions but in that case you colour access it this way directly:

- `colour.RGB_COLOURSPACES['ProPhoto RGB']`
- `colour.models.PROPHOTO_RGB_COLOURSPACE`

Thus to convert from ProPhoto RGB to sRGB you would just do that:

```python
colour.RGB_to_RGB(
    my_array,
    colour.RGB_COLOURSPACES['ProPhoto RGB'],
    colour.RGB_COLOURSPACES['sRGB'])
```

Cheers,

Thomas

**Comment by hminle** [comment]
Hi @KelSolaar ,
Thank for your guide
I just wonder: in the RGB_to_RGB function, the output is also based on chromatic_adaptation_transform, right?

**Comment by hminle** [comment]
Hi @KelSolaar ,
I've just tried your way.
But there're still some points outside the gamut region for sRGB.
![image](https://user-images.githubusercontent.com/16201681/74035404-331add80-4988-11ea-8f2e-bc6a18c1592f.png)

Best,
Hoang

**Comment by thomasmansencal** [comment]
It is entirely possible that ProPhoto RGB colours are out of sRGB Gamut, the former is much larger than the latter.

**Comment by hminle** [comment]
Hi @KelSolaar 
So you guys do not apply any Gamut Mapping Algorithm, right?
(https://graphics.stanford.edu/courses/cs178/applets/gamutmapping.html)

Best,
Hoang

**Comment by thomasmansencal** [comment]
Unfortunately no, but I have started to look at that seriously here: https://github.com/colour-science/colour/tree/feature/gamut_mapping, and related issues are #154 and #311.

**Comment by sobotka** [comment]
Something about that plot is fishy; the chromaticities shouldn’t be clustered along the edge of the locus. Something seems odd there?

**Comment by hminle** [comment]
> Something about that plot is fishy; the chromaticities shouldn’t be clustered along the edge of the locus. Something seems odd there?

@sobotka , as KelSolaar mentioned earlier, they are investigating about the gamut mapping.

**Comment by sobotka** [comment]
It’s unrelated to gamut mapping.

The chromaticities in the source to destination will remain in relative positions. The issue is the collapsing of the chromaticities along that right edge of the spectral locus.

Is it possible that the ProPhoto image was clipped to ProPhoto?

**Comment by hminle** [comment]
Hi @sobotka,
I got the image from this link: https://data.csail.mit.edu/graphics/fivek/
The image ID: 0001, expert A.
You can download and check yourself.
The expert edits the raw DNG image using Photoshop and save it with ProPhoto RGB.

I think this process may have some effects on color chromaticities.

Best

**Comment by nick-shaw** [comment]
I think it may be related to the use of unlinearised data. `RGB_to_RGB` expects and produces linear data unless `apply_cctf_decoding` and `apply_cctf_encoding` are set to `True`. Likewise I believe the plotting functions expect linear data.
